modules/UpgradeWizard/uw_utils.php

```diff
@@ -2248,42 +2256,54 @@ function unlinkUploadFiles()
 //	}
 }
 
+/**
+ * Recursively deletes a directory tree.
+ *
+ * @param string $folder         The directory path.
+ * @param bool   $keepRootFolder Whether to keep the top-level folder.
+ *
+ * @return bool TRUE on success, otherwise FALSE.
+ */
+function deleteTree($folder, $keepRootFolder = false) {
+	// Handle bad arguments.
+	if (empty($folder) || !file_exists($folder)) {
+		return true; // No such file/folder exists.
+	} elseif (is_file($folder) || is_link($folder)) {
```

Comment:
This can be converted into an if branch for lower complexity:

```php
    // Handle bad arguments.
    if (empty($folder) || !file_exists($folder)) {
        // No such file/folder exists.
        return true;
    }

    if (is_file($folder) || is_link($folder)) {
        // Delete file/link.
        return @unlink($folder);
    }
```